corehq/motech/models.py

```diff
@@ -153,6 +153,10 @@ def get_requests(
         )
 
     def get_auth_manager(self):
+        # FIXME this may construct an auth manager with a null username.
+        # This can result in an auth header with a username of "None",
+        # which is almost certainly not correct. Basic and Digest auth
+        # are affected, and possibly others.
```

Comment:
@kaapstorm FYI, more context can be found in commit https://github.com/dimagi/commcare-hq/pull/31482/commits/e52ea875ab36c59b8f5a70d7219846da2e66605d. Is this something you could fix?

Reply:
[OAuth1Manager](https://github.com/dimagi/commcare-hq/blob/99aa9dcf4e23bde6223f9f27906f8ab5a369dd2a/corehq/motech/auth.py#L125) and [OAuth2ClientGrantManager](https://github.com/dimagi/commcare-hq/blob/99aa9dcf4e23bde6223f9f27906f8ab5a369dd2a/corehq/motech/auth.py#L174) don't take a `username` parameter. For auth managers that do take `username`, we could assert that it has a value in the constructor. Does that sound right?

Reply:
Possibly. I think the problem is that `ConnectionSettings` can exist in a state that should not be allowed (null username with incompatible auth type). One way to prevent this would be to add a [check constraint](https://www.postgresql.org/docs/9.4/ddl-constraints.html#DDL-CONSTRAINTS-CHECK-CONSTRAINTS) to either the `auth_type` or `username` field, disallowing a null value in `username` when `auth_type` is either `'basic'` or `'digest'` (and possibly other auth types that require a username). That way a `ConnectionSettings` row could not be saved in an invalid state.

A less correct option would be to `return AuthManager()` here if the auth type does not support a null username and `self.username is None`. This option does not require that all existing invalid data states be cleaned up, and will continue to allow invalid connection settings configurations to be saved in the database.

Reply:
https://github.com/dimagi/commcare-hq/pull/31487

Reply:
(Sorry, I wrote that before reading your comment. Kinda wish GitHub loaded comments without requiring a refresh. I agree that preventing invalid ConnectionSettings instances is better. I'll create a ticket for that, and follow up.)